### ISSUE TITLE
[sort-comp] static getter should be treated as static method over common getter

### ISSUE THREAD

**Issue by kouhin** [yannickcr/eslint-plugin-react#1808]
Configuration (airbnb):

```js
{
  rules: {
    'react/sort-comp': ['error', {
      order: [
        'static-methods',
        'instance-variables',
        'lifecycle',
        '/^on.+$/',
        'getters',
        'setters',
        '/^(get|set)(?!(InitialState$|DefaultProps$|ChildContext$)).+$/',
        'instance-methods',
        'everything-else',
        'rendering',
      ],
  }
}
```

Here is a simple example: 

```jsx
import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

export class Icon extends React.Component {
  static propTypes = {
    name: PropTypes.string,
  };

  static get defaultProps() { // LINT ERROR
    return {
      name: 'defaultIcon',
    };
  }

  shouldComponentUpdate(nextProps) {
    return this.props.name === nextProps.name;
  }

  render() {
    return (
      <i // eslint-disable-line jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events
        className={cx('s', `s-${this.props.name}`)}
      />
    );
  }
}
```

It causes an error on `static get defaultProps()` and error message is `getter functions should be placed after shouldComponentUpdate [react/sort-comp]`.

**Comment by ljharb** [comment]
This is correct; also, using static getters like this is a big performance hit. Use `Icon.defaultProps = ` instead.

**Comment by kouhin** [comment]
@ljharb Thank you. Actually, we are using `static get defaultProps()` in our project in order to avoid memory leak. 
However, besides performance problem of this pattern (maybe anti-pattern), I think `static getters` should be treated as `static` or `static getters` rather than common getters.

**Comment by ljharb** [comment]
I'm not sure how there would be a memory leak from an object created once and assigned as a property.

It might make sense to make a separate category for "static getters", but it's absolutely an antipattern to use them in the first place.
